### PR TITLE
Create Course: Filter out unversioned families

### PR DESCRIPTION
**Pull request by dmcavoy**
When creating a course (unit group or standalone unit) you are asked to pick or input a family name. The current dropdown lists all the family names available for that type of course (unit group or standalone unit). However we have families where there are one course in it and its version year is `unversioned` for those families we do not want to add other versions to the family so this PR removes the family names with unversioned courses in them from the list. 

This is follow up work to make sure:
- You can't input an exist family name
- Update family names to not end in a year (self-paced-pl-csd-2021 for example)

## Links

https://codedotorg.atlassian.net/browse/PLAT-1514

## Testing story

- Tested it out locally
Current Levelbuilder - New Script:

Screen Shot 2022-05-06 at 1 25 34 PM

Local - New Script:

Screen Shot 2022-05-06 at 1 33 05 PM

Current Levelbuilder - New Course

Screen Shot 2022-05-06 at 1 33 55 PM

Local - New Course

Screen Shot 2022-05-06 at 1 34 19 PM

### CODE REVIEW

**Comment by dmcavoy** [comment]
Hold on. Found something I need to check on

**Comment by dmcavoy** [comment]
Alright. Ready for review now.

File: dashboard/app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -16,10 +16,17 @@ class CoursesController < ApplicationController
...
+    @course_families_course_types = []
+    UnitGroup.family_names.map do |cf|
+      co = CourseOffering.find_by(key: cf)
+      first_cv = co.course_versions.first
+      ug = first_cv.content_root
+      @versioned_course_families << cf unless first_cv.key == 'unversioned'
+      @course_families_course_types << [cf, {instruction_type: ug.instruction_type, instructor_audience: ug.instructor_audience, participant_audience: ug.participant_audience}]
+    end
+
+    @course_families_course_types.to_h
```
Comment (davidsbailey) [inline dashboard/app/controllers/courses_controller.rb:29]: can you remind me, does the return value for this method get used somewhere? here and below. 

the place I would start looking is https://guides.rubyonrails.org/action_controller_overview.html, but I didn't see anything definitive. if you can't find a way that it's used, and you feel good about your test coverage, I would suggest removing it.

**Review by davidsbailey — APPROVED** [review]
Looks great Dani! In addition to keeping us out of trouble, I love how much shorter these lists are now! much easier to find what you're actually looking for 🔎

File: dashboard/app/controllers/courses_controller.rb (hunk 1 above)
Comment (dmcavoy) [reply dashboard/app/controllers/courses_controller.rb:29]: I wasn't actually trying to return `@course_families_course_types` I just needed for follow up after setting it up to convert it to a hash. I'm going to leave this and merge it but if you have thoughts when you get back I'm happy to follow up

Comment (davidsbailey) [reply dashboard/app/controllers/courses_controller.rb:29]: This returns a hash, but does not convert the originals object into a hash, so this statement ultimately looks like it has no effect.

Comment (dmcavoy) [reply dashboard/app/controllers/courses_controller.rb:29]: Oh I see what I missed. Ok thanks I will look into this.

Comment (dmcavoy) [reply dashboard/app/controllers/courses_controller.rb:29]: Fixing it here: https://github.com/code-dot-org/code-dot-org/pull/46239

**Pull request merged**